src/ActionList/ActionList.features.stories.tsx

```diff
@@ -96,6 +96,146 @@ export const WithVisualListHeading = () => (
   </ActionList>
 )
 
+export const ListWithNewGroupHeadingAPI = () => (
```

Comment:
From a documentation perspective, would it help to have the story name be more like `ListWithHeading` or something similar? Trying to think of when someone might be looking for how to do X (where X is adding a heading for a list) and having a story name that makes sense to them.

The terminology "NewGroupedHeadingAPI" feels very clear for maintainers but from a user perspective might not make as much sense.

Reply:
I'm going to take these stories back in this PR. I am thinking to do a follow up one that adds a `@deprecated` notation to the `title` prop and update all title usages with the new API in stories and docs. Let me know if there is any concern!